webkitbot should accept commit identifier

JonWBedard opened the pull request:
#### bb9553870ebc05557accce6781852762de4fef01
<pre>
webkitbot should accept commit identifier
<a href="https://bugs.webkit.org/show_bug.cgi?id=224063">https://bugs.webkit.org/show_bug.cgi?id=224063</a>
&lt;rdar://76114559&gt;

Reviewed by Stephanie Lewis.

* Tools/Scripts/webkitpy/common/checkout/checkout.py:
(_changelog_data_for_revision): Support any commit argument.
* Tools/Scripts/webkitpy/common/checkout/checkout_mock.py:
(MockCheckout.commit_info_for_revision): Support r prefixed strings.
* Tools/Scripts/webkitpy/common/checkout/scm/git.py:
(Git.changed_files_for_revision): Use git_commit_from_string.
(Git.git_commit_from_string): Renamed from git_commit_from_svn_revision.
(Git.contents_at_revision): Use git_commit_from_string.
(Git.diff_for_revision): Ditto
(Git.committer_email_for_revision): Ditto.
(Git.apply_reverse_diff): Ditto.
(Git.git_commit_from_svn_revision): Rename git_commit_from_string.
* Tools/Scripts/webkitpy/common/config/urls.py:
(view_revision_url): Support any commit argument.
* Tools/Scripts/webkitpy/common/net/bugzilla/attachment.py:
(Attachment): Caller should prefix r, if appropriate.
* Tools/Scripts/webkitpy/tool/commands/download.py:
(AbstractRevertPrepCommand._prepare_state): Handle multiple types of commit arguments.
(CreateRevert._prepare_state): Remove r prefix.
* Tools/Scripts/webkitpy/tool/commands/download_unittest.py:
(AbstractRevertPrepCommandTest.test_prepare_state):
* Tools/Scripts/webkitpy/tool/steps/reopenbugafterrevert.py:
(ReopenBugAfterRevert.run): Remove r prefix.
* Tools/WebKitBot/src/WebKitBot.mjs: Allow hashes and identifiers in webkitbot.

Canonical link: <a href="https://commits.webkit.org/252075@main">https://commits.webkit.org/252075@main</a>
</pre>

sundiamonde commented on Tools/Scripts/webkitpy/common/checkout/checkout_mock.py:
```diff
@@ -104,6 +104,11 @@ def __init__(self):
+        if isinstance(svn_revision, str) and svn_revision.startswith('r'):
+            svn_revision = svn_revision[1:]
+        if isinstance(svn_revision, str) and not svn_revision.isdigit():
+            return None
```
why none here?

sundiamonde replied:
or more to the point will this fail silently or with a weird error message

JonWBedard replied:
This is a mock class. There is already a comment in this function that says this:
```
# any "unrecognized" svn_revision will return None.
```
just wasn't in the diff.

sundiamonde commented on Tools/Scripts/webkitpy/common/net/bugzilla/attachment.py:
```diff
@@ -37,8 +37,8 @@
 class Attachment(object):
     fast_cq_preamble = "[fast-cq] "
-    revert_preamble = "REVERT of r"
-    rollout_preamble = "ROLLOUT of r"
```
do we need to append r to svn_revisions somewhere?

JonWBedard replied:
We already do in [Tools/Scripts/webkitpy/tool/commands/download.py](https://github.com/WebKit/WebKit/pull/1992/files#diff-5577606b5e5254a7108d3e3c86dcbeb56f4e53c74f2f5e0dd60245d44dd2dbd4)

```
revision_list = ['r{}'.format(rev) for rev in revisions] + commits
```

commits will contain hashes and identifiers, so doesn't need the same treatment.

JonWBedard committed:
webkitbot should accept commit identifier
https://bugs.webkit.org/show_bug.cgi?id=224063
<rdar://76114559>

Reviewed by Stephanie Lewis.

* Tools/Scripts/webkitpy/common/checkout/checkout.py:
(_changelog_data_for_revision): Support any commit argument.
* Tools/Scripts/webkitpy/common/checkout/checkout_mock.py:
(MockCheckout.commit_info_for_revision): Support r prefixed strings.
* Tools/Scripts/webkitpy/common/checkout/scm/git.py:
(Git.changed_files_for_revision): Use git_commit_from_string.
(Git.git_commit_from_string): Renamed from git_commit_from_svn_revision.
(Git.contents_at_revision): Use git_commit_from_string.
(Git.diff_for_revision): Ditto
(Git.committer_email_for_revision): Ditto.
(Git.apply_reverse_diff): Ditto.
(Git.git_commit_from_svn_revision): Rename git_commit_from_string.
* Tools/Scripts/webkitpy/common/config/urls.py:
(view_revision_url): Support any commit argument.
* Tools/Scripts/webkitpy/common/net/bugzilla/attachment.py:
(Attachment): Caller should prefix r, if appropriate.
* Tools/Scripts/webkitpy/tool/commands/download.py:
(AbstractRevertPrepCommand._prepare_state): Handle multiple types of commit arguments.
(CreateRevert._prepare_state): Remove r prefix.
* Tools/Scripts/webkitpy/tool/commands/download_unittest.py:
(AbstractRevertPrepCommandTest.test_prepare_state):
* Tools/Scripts/webkitpy/tool/steps/reopenbugafterrevert.py:
(ReopenBugAfterRevert.run): Remove r prefix.
* Tools/WebKitBot/src/WebKitBot.mjs: Allow hashes and identifiers in webkitbot.

Canonical link: https://commits.webkit.org/252075@main
